Improve application documentation (#1587)
chaos-polymtl/lethe: doc/source/applications.rst

Applications Overview
######################

> Lethe has several applications that can be used depending on the problem to be solved. Once Lethe is compiled, all the executables to these applications become available. In the following table, a brief description of each application is provided:

Author: Yeah that's a valid point, I won't mark the deprecation but I can reorder them from most common to more specialized let's say.

.. list-table::
    :header-rows: 1
    :widths: 40 40 60 

    * - Physics 
      - Application

>       - Description

Author: Ohhh I like it... 
I'll switch it to be this way.

    * - * Single-phase flows

>         * Fluid-fluid flows 

Reviewer: You can also use Free surface flows, but I like fluid-fluid flows!

        * Heat transfer
        * Tracer
      - ``lethe-fluid``

>       - This application solves the Navier-Stokes equations in a monolithic way. **This is the most robust fluid dynamics solver in Lethe  and it is compatible with all multiphysics features**.

Reviewer: ```suggestion
      - This application solves the Navier-Stokes equations in a monolithic way. **This is the most robust fluid dynamics solver in Lethe and it is compatible with all multiphysics features**.
```

    * - * Single-phase flows
        * Heat transfer
        * Tracer
      - ``lethe-fluid-matrix-free``

>       - This application solves the Navier-Stokes equations in a monolithic way and using a matrix-free approach.  **This is the fastest fluid dynamics solver in Lethe. This solver should be used for high-order simulations. This solver does not fully support all multiphysics features**.

Reviewer: ```suggestion
      - This application solves the Navier-Stokes equations in a monolithic way and using a matrix-free approach.  **This is the fastest fluid dynamics solver in Lethe. It should be used for high-order simulations. It does not fully support all multiphysics features**.
```

    * - * Single-phase flows
        * Fluid-fluid flows 
        * Heat transfer
        * Tracer

>         * Immersed boundaries

Reviewer 2: I find it weird to have Immersed boundaries be in `Physics`, maybe `Features` ?

Author: Yeah that's a good point. But then I don't necessary want to add another column...
Maybe it could be single-phase flows with immersed geometries?

Reviewer 2: Yes I like it!

      - ``lethe-fluid-nitsche``
      - This application solves the Navier-Stokes equations by imposing immersed boundary conditions without discretizing the boundaries using the Nitsche method. 
    * - * Particle-laden flows
      - ``lethe-fluid-particles``
      - This application allows to run unresolved Computational Fluid Dynamics-Discrete Element Method simulations.

>     * - * Particle-laden flows

Reviewer 2: ```suggestion
    * - * Single-phase flows
        * Particle-laden flows
```
Technically `lethe-fluid-sharp` can do single-phase flows with immersed boundaries (Same as Nitsche impeller), but it depends how we wish to sell the use of this solver.

        * Immersed boundaries
      - ``lethe-fluid-sharp``

>       - This application solves the Navier-Stokes equations by imposing immersed boundary conditions without discretizing the boundaries using the sharp-edge method. It is used for resolved Computational Fluid Dynamics-Discrete Element Method simulations.

Reviewer: ```suggestion
      - This application solves the Navier-Stokes equations for particle-laden flows. A sharp-edge method is used to impose immersed boundary conditions at the particle walls, removing the requirement of a boundary-conforming mesh. This application allows to perform resolved Computational Fluid Dynamics-Discrete Element Method simulations.
```

    * - * Single phase volume-averaged flows
      - ``lethe-fluid-vans``

>       - This application solves the Volume-Averaged Navier-Stokes equations in a monolithic way.

Reviewer: For me, it is unclear the difference between `lethe-fluid-particles` and `lethe-fluid-vans`.

    * - * Granular flow of spherical particles
      - ``lethe-particles``
      - This application uses the Discrete Element Method to simulate spherical particles. 

>     * - * Single-phase flows

Reviewer: ```suggestion
    * - * Steady-state single-phase flows
```
